XRTK-Core/Packages/com.xrtk.core/Providers/Controllers/Simulation/Hands/SimulatedHandControllerState.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using UnityEngine;
+using XRTK.Definitions.Controllers.Simulation;
+using XRTK.Definitions.Utilities;
+using XRTK.Providers.Controllers.Hands;
+using XRTK.Services;
+
+namespace XRTK.Providers.Controllers.Simulation.Hands
+{
+    /// <summary>
+    /// Internal class to define current hand data and smoothly animate hand data points.
+    /// </summary>
+    public class SimulatedHandControllerState
```

Comment:
I'm getting confused with a lot of these classes. Many of them seem to be doing similar things.

Reply:
The naming can be confusing at times. I've been through that. But each of them is responsible for a specific part. That way 1000+ lines classes are avoided.